src/CSharpExtDownloader.ts

```diff
@@ -0,0 +1,148 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as vscode from 'vscode';
+import TelemetryReporter from 'vscode-extension-telemetry';
+import * as util from './common';
+import { Logger } from './logger';
+import { PackageManager, Status, PackageError } from './packages';
+import { PlatformInformation } from './platform';
+import { VSCodePlatformInformation } from './vscodePlatform';
+
+/*
+ * Class used to download the runtime dependencies of the C# Extension
+ */
+export class CSharpExtDownloader
+{
+    public constructor(
+        private channel: vscode.OutputChannel,
+        private logger: Logger,
+        private reporter: TelemetryReporter /* optional */,
+        private packageJSON: any) {
+    }
+
+    public installRuntimeDependencies(): Promise<boolean> {
+        this.logger.append('Updating C# dependencies...');
+        this.channel.show();
+
+        let statusItem = vscode.window.createStatusBarItem(vscode.StatusBarAlignment.Right);
+        let status: Status = {
+            setMessage: text => {
+                statusItem.text = text;
+                statusItem.show();
+            },
+            setDetail: text => {
+                statusItem.tooltip = text;
+                statusItem.show();
+            }
+        };
+
+        // Sends "AcquisitionStart" telemetry to indicate an acquisition  started.
+        if (this.reporter) {
+            this.reporter.sendTelemetryEvent("AcquisitionStart");
+        }
+
+        let platformInfo: PlatformInformation;
+        let packageManager: PackageManager;
+        let installationStage = 'touchBeginFile';
+        let errorMessage = '';
+        let success = false;
+
+        let telemetryProps: any = {};
+
+        return util.touchInstallFile(util.InstallFileType.Begin)
+            .then(() => {
+                installationStage = 'getPlatformInfo';
+                return VSCodePlatformInformation.GetCurrent();
+            })
+            .then(info => {
+                platformInfo = info;
+                packageManager = new PackageManager(info, this.packageJSON);
+                this.logger.appendLine();
+
+                // Display platform information and RID followed by a blank line
+                this.logger.append(`Platform: ${info.toString()}`);
+                if (info.runtimeId) {
+                    this.logger.appendLine(` (${info.runtimeId})`);
+                }
+                else {
+                    this.logger.appendLine();
```

Comment:
That sounds good to me. One thing that's also been a small point of confusion is that this 'runtimeId' isn't _really_ the current runtimeId. It's the effective runtimeId to locate the correct package. For example, on macOS Sierra, it reports osx.10.11-x64 instead of osx.10.12-x64. Should we tweak the text to make it clearer?